frappe/email/utils.py

```diff
@@ -16,3 +18,25 @@ def get_port(doc):
 			doc.incoming_port = poplib.POP3_SSL_PORT if doc.use_ssl else poplib.POP3_PORT
 
 	return cint(doc.incoming_port)
+
+
+def decode_sequence(encoded_sequence) -> str:
+	"""
+	Decodes a encoded_sequence consisting of a tuple (string, charset_encoding). The function concatenates all chunks and returns the resulting decoded string.
+	Args:
+		encoded_sequence ((string, charset_encoding)): A list of tuples where each tuple contains a chunk of the string and its encoding.
+	Returns:
+		str: The decoded and concatenated sequence string.
+	"""
+	from frappe import safe_decode
+	decoded_string = ""
+	for chunk, encoding in encoded_sequence:
+		decoded_chunk = ""
+		if not isinstance(chunk, str):
```

Comment:
Maybe it is necessary to reduce the redundant code to this?
```
		if not isinstance(chunk, str):
			detected_encoding = encoding if encoding is not None and encoding != 'unknown-8bit' else chardet.detect(chunk)["encoding"]
			decoded_string += chunk.decode(detected_encoding, 'ignore')
		else:
			decoded_string += chunk
```
1. If  type(chunk) = str, then it has already been decoded and it is not worth trying to decode it again.
2. Use condition:
```
		if isinstance(decoded_chunk, str):
			decoded_string += decoded_chunk
```
not the best option, because in case of a decoding error, we will completely lose the entire chunk. I think it would be better to use **chunk.decode(detected_encoding, 'ignore')** instead of **safe_decode(param=chunk, encoding=detected_encoding)** so most of the chunk will be decoded correctly.

Reply:
???

Reply:
Am I wrong about something?